pkg/populator/endpoint.go

```diff
@@ -0,0 +1,70 @@
+package populator
+
+import (
+	"os"
+	"path/filepath"
+	"sync/atomic"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+type EndpointPopulator struct {
+	// kubeConfigFile is the path where the kube config is stored
+	// Only access this with atomic ops
+	kubeConfigFile atomic.Value
+}
+
+func (e *EndpointPopulator) SetKubeConfigFile() {
+	var kubeConfigFile string
+	if os.Getenv("KUBECONFIG") == "" {
+		hdir, err := homedir.Dir()
```

Comment:
I think finding the kubeconfig file not should be part of the Populators responsibility, maybe pass the path to the constructor or as part of calling the PopulateKubeConfig